src/Domain/Beginner.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace PhpTuf\ComposerStager\Domain;
+
+use PhpTuf\ComposerStager\Exception\DirectoryAlreadyExistsException;
+use PhpTuf\ComposerStager\Exception\DirectoryNotFoundException;
+use PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem;
+use PhpTuf\ComposerStager\Infrastructure\Process\FileCopier;
+
+class Beginner implements BeginnerInterface
+{
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Process\FileCopier
+     */
+    private $fileCopier;
+
+    /**
+     * @var \PhpTuf\ComposerStager\Infrastructure\Filesystem\Filesystem
+     */
+    private $filesystem;
+
+    public function __construct(FileCopier $fileCopier, Filesystem $filesystem)
+    {
+        $this->fileCopier = $fileCopier;
+        $this->filesystem = $filesystem;
+    }
+
+    public function begin(string $activeDir, string $stagingDir, ?callable $callback = null): void
+    {
+        if (!$this->activeDirectoryExists($activeDir)) {
+            throw new DirectoryNotFoundException($activeDir, 'The active directory does not exist at "%s"');
```

Comment:
This feels very much like repeating the checks in BeginCommand. Why not simply call `begin()` and catch these exceptions, instead of explicitly checking twice?

Reply:
Mmm... I wasn't sure that was the right separation of responsibilities at first, but you're probably right. Done.